doc/cfg/src/cfg.lalrpop

```diff
@@ -0,0 +1,85 @@
+grammar;
+
+pub AlwaysCreated: () = {
+};
+
+#[cfg(feature = "test-not-set")]
+pub NotCreated: () = {
+};
+
+#[cfg(feature = "test-set")]
+pub Created: () = {
+    "x" => ()
+};
+
+#[cfg(not(feature = "test-not-set"))]
+pub CreatedWithNot: () = {
+    "x" => ()
+};
+
+#[cfg(not(feature = "test-set"))]
+pub NotCreatedWithNot: () = {
+    "x" => ()
+};
+
+#[cfg(any(feature = "test-not-set", feature = "test-set"))]
+pub CreatedWithAny: () = {
+    "x" => ()
+};
+
+#[cfg(any(feature = "test-not-set"))]
+pub NotCreatedWithAny: () = {
+    "x" => ()
+};
+
+#[cfg(all(feature = "test-set"))]
+pub CreatedWithAll: () = {
+    "x" => ()
+};
+
+#[cfg(all(feature = "test-set", feature = "test-not-set"))]
+pub NotCreatedWithAll: () = {
+    "x" => ()
+};
```

Comment:
Given that we've added the restriction to only have test-set or test-not-set, is the following no longer a meaningful case since it will never get created? It's not clear to me that we can test the absence of this case in general.

Reply:
One of [these](https://github.com/rust-lang/cargo/issues/1194#issuecomment-1482596022) could be an option...

Kind of hacky, but we could also just declare a type called `NotCreatedWithAllParser` at a point where it would name-collide with the generated one if it existed?

The whole thing is a little weird, since lalrpop will just fail to build the grammar with both features on, due to the `TwiceMacro` below.  Maybe what we want is a third feature that is compatible with `test-set` like `test-set2`, since otherwise we're not testing `all()` with multiple features I think.